Es4 discovery library - Revamp

mickey4653 opened the pull request:
# Description
Revamped Dicovery library Feature

Ticket:
- created a component for discovery library feature
- integrated the discovery library feature
- created a discovery library window
- added images necessary for discovery library
- integrated discovery library via the main layout to follow code pattern in the project
- created custom prompts in the constants folder
	
# How Has This Been Tested?
- Video Link: https://jam.dev/c/0a55882f-0af1-4f61-ba67-72ab67d64fe8

mickey4653 committed:
Made Some Changes:
	- created a component for discovery library feature
	- integrated the discovery library feature
	- created a discovery library window
	- added images necessary for discovery library
	- integrated discovery library via the main layout to follow code pattern in the project
	- created custom prompts in the constants folder

mickey4653 committed:
Made Some Changes:
	- adjusted styling to make discovery library feature fit well along side chat interface

bkb-Git commented on .firebaserc:
```diff
@@ -1,6 +1,7 @@
   "projects": {
-    "default": "kai-ai-b8ac9"
+    "default": "kai-ai-b8ac9",
+    "mickey's-app-firebase": "kai-platform-b53aa"
```
remove this

bkb-Git commented on frontend/constants/prompts.js:
```diff
@@ -0,0 +1,25 @@
+const categorizePrompts = [
```
Let's create a collection called "personas" and create the db schema for this collection and fetch them when the user loads the dicovery library.

bkb-Git commented on frontend/layouts/MainAppLayout/ MainAppLayout.jsx:
```diff
@@ -49,12 +56,25 @@ const MainAppLayout = (props) => {
+        />
         <Grid {...styles.contentGridProps(extraContentProps, isToolPage)}>
           {children}
+          {/* DiscoveryLibraryWindow component displays a sidebar that contains discovery Library. This component is rendered on the right side of the chat interface. */}
```
Remove commented code

bkb-Git commented on frontend/layouts/MainAppLayout/NavBar/Navbar.jsx:
```diff
@@ -126,7 +131,7 @@ const NavBar = () => {
 
       // TODO: Once Discovery Feature is ready, uncomment below statement.
-      // if (id === 'discovery') return discoveryRegex.test(pathname);
+      if (id === 'discovery') return discoveryRegex.test(pathname);
```
Let's not directly use strings like above, store these literals in a constant file and reference them here

`${ROUTES_IDS.DISCOVERY}`

bkb-Git commented on frontend/pages/discovery/index.jsx:
```diff
@@ -0,0 +1,13 @@
+import DiscoveryLibrary from '@/templates/Chat/DiscoveryLibrary';
+
+const MarvelDiscovery = () => {
+  localStorage.removeItem('sessionId');
```
you can remove the above line of code, don't think it's necessary in this page

bkb-Git commented on frontend/templates/Chat/DiscoveryLibrary/DiscoveryLibrary.jsx:
```diff
@@ -0,0 +1,110 @@
+
+  useEffect(() => {
+    // Define your custom prompts here
+
```
Remove above line

bkb-Git commented on frontend/templates/Chat/DiscoveryLibrary/DiscoveryLibrary.jsx:
```diff
@@ -0,0 +1,110 @@
+          <Grid item {...styles.avatarImageGridProps}>
+            <AvatarImage {...styles.avatarImageProps} />
+            {/* <img src={<AvatarImage/>} alt="Avatar"/> */}
+            {/* {console.log("Avatar image present: ",<AvatarImage/>)} */}
```
remvoe comments

bkb-Git commented on frontend/templates/Chat/DiscoveryLibrary/styles.js:
```diff
@@ -0,0 +1,272 @@
+    mobileSmall: true,
+    position: 'absolute',
+    top: -80,
+    // left: 20,
```
remove commented code

bkb-Git commented on frontend/templates/Chat/DiscoveryLibraryWindow/DiscoveryLibraryWindow.jsx:
```diff
@@ -0,0 +1,391 @@
+    }
+    return null;
+  };
+  const renderStartChatMessage = () => {
```
add a new line above

bkb-Git commented on frontend/templates/Chat/DiscoveryLibraryWindow/DiscoveryLibraryWindow.jsx:
```diff
@@ -0,0 +1,391 @@
+          onScroll={handleOnScroll}
+          {...styles.centerChat.messagesGridProps}
+        >
+          {/* Render the start chat message if there are no chat messages or if the info chat is not open. */}
```
remove commented code

bkb-Git commented on frontend/templates/Chat/DiscoveryLibraryWindow/DiscoveryLibraryWindow.jsx:
```diff
@@ -0,0 +1,391 @@
+          {/* Render the start chat message if there are no chat messages or if the info chat is not open. */}
+          {(chatMessages?.length === 0 || !chatMessages) && !infoChatOpened
+            ? renderStartChatMessage()
+            : null}
```
You can just write:

`(chatMessages?.length === 0 || !chatMessages) && !infoChatOpened && renderStartChatMessage()`

bkb-Git commented on frontend/templates/Chat/DiscoveryLibraryWindow/DiscoveryLibraryWindow.jsx:
```diff
@@ -0,0 +1,391 @@
+      </Grid>
+    );
+  };
+  const renderNewMessageIndicator = () => {
```
add empty line above

bkb-Git commented on frontend/templates/Chat/DiscoveryLibraryWindow/DiscoveryLibraryWindow.jsx:
```diff
@@ -0,0 +1,391 @@
+      return (
+        <Grid {...styles.bottomChatContent.bottomChatContentGridProps}>
+          {/* Default Prompt Component */}
+          {/* <DefaultPrompt handleSendMessage={handleSendMessage} /> */}
```
remvoe commented code

bkb-Git commented on frontend/templates/Chat/DiscoveryLibraryWindow/DiscoveryLibraryWindow.jsx:
```diff
@@ -0,0 +1,391 @@
+          {/* Default Prompt Component */}
+          {/* <DefaultPrompt handleSendMessage={handleSendMessage} /> */}
+
+          {/* Quick Actions Component */}
```
same here

bkb-Git commented on frontend/templates/Chat/DiscoveryLibraryWindow/DiscoveryLibraryWindow.jsx:
```diff
@@ -0,0 +1,391 @@
+
+  return (
+    <Grid container {...styles.chatInterface}>
+      {/* Discovery Library Slide */}
```
remove commented code

bkb-Git commented on frontend/templates/Chat/DiscoveryLibraryWindow/DiscoveryLibraryWindow.jsx:
```diff
@@ -0,0 +1,391 @@
+        {/* Center Chat Content */}
+        {renderCenterChatContent()}
+        {renderNewMessageIndicator()}
+        {/* Bottom Chat Content */}
```
same here as well

bkb-Git commented on frontend/templates/Chat/DiscoveryLibraryWindow/styles.js:
remove commented code

bkb-Git reviewed (changes requested):
Looks mostly good, just need to avoid pushing code with comments in them and also let's try and make the code readable by adding whitespacing where necessary.

mickey4653 committed:
Made Review Changes:
	- change project name to default
	- modifed constants for prompts to personas
	- added route_ids file under constants dir
	- removed comments
	- created database collection for personas
	- added thunks and slices for personas and made sure its been stored in redux

mickey4653 committed:
Made Extra Review Changes:
	- removed comments
	- added new lines in specified areas
	- refractored code in discovery library window

mickey4653 committed:
fixes:
	- eslint fix

mickey4653 committed:
- removed comment

mickey4653 committed:
- removed extra comments
